pkg/infradb/subscriberframework/eventbus/event_bus_test.go

```diff
@@ -0,0 +1,181 @@
+package eventbus
+
+import (
+	"log"
+	"sync"
+	"testing"
+	"time"
+)
+
+type ModulelciHandler struct {
+	receivedEvents []*ObjectData
+	sync.Mutex
+}
+
+func (h *ModulelciHandler) HandleEvent(eventType string, objectData *ObjectData) {
+	h.Lock()
+	defer h.Unlock()
+	h.receivedEvents = append(h.receivedEvents, objectData)
+	switch eventType {
+	case "testEvent":
+	case "testEventpriority":
+	case "testEventChBusy":
+	case "testEventUnsub":
+		log.Printf("received event type %s", eventType)
+	default:
+		log.Printf("error: Unknown event type %s", eventType)
+	}
+}
+
+func TestSubscribeAndPublish(t *testing.T) {
+	handler := &ModulelciHandler{}
+
+	EventBus := NewEventBus()
+	EventBus.StartSubscriber("testModule", "testEvent", 1, handler)
+
+	time.Sleep(100 * time.Millisecond)
+
+	objectData := &ObjectData{
+		ResourceVersion: "v1",
+		Name:            "testObject",
+		NotificationID:  "123",
+	}
+
+	subscribers := EventBus.GetSubscribers("testEvent")
+	if len(subscribers) == 0 {
+		t.Errorf("No subscribers found for event type 'testEvent'")
+	}
+	subscriber := subscribers[0]
+
+	err := EventBus.Publish(objectData, subscriber)
+	if err != nil {
+		t.Errorf("Publish() failed with error: %v", err)
+	}
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
by doing that we are kind of diverting the whole test. then it will never fail if there is any change in task manager code.

Reply:
I do not understand...
`HandleEvent` is part of a test, and we are free to organize it as we need
The current execution time of the test is not appropriate: [159.629s](https://github.com/opiproject/opi-evpn-bridge/actions/runs/11249540919/job/31276636859?pr=391#step:5:296) and I think we could do better here instead of using sleeps